src/common/src/util/ordered/serde.rs

```diff
@@ -154,32 +153,27 @@ pub fn serialize_pk_and_row(
                     // we serialize this null row specially by only using one cell encoding.
                 }
                 datum => {
-                    all_null = false;
                     let value = serialize_cell(datum)?;
                     result.push((key, Some(value)));
                 }
             },
             None => {
                 // A `None` of row means deleting that row, while the a `None` of datum represents a
                 // null.
-                all_null = false;
                 result.push((key, None));
             }
         }
     }
-    if all_null {
-        // Here we use a special column id -1 to represent a row consisting of all null values.
-        // `MViewTable` has a `get` interface which accepts a cell id. A null row in this case
-        // would return null datum as it has only a single cell with column id == -1 and `get`
-        // gets nothing.
-        let key = [
-            pk_buf,
-            serialize_column_id(&NULL_ROW_SPECIAL_CELL_ID)?.as_slice(),
-        ]
-        .concat();
+
+    if row.is_none() {
+        let key = [pk_buf, serialize_column_id(&SENTINEL_CELL_ID)?.as_slice()].concat();
+        result.push((key, None));
```

Comment:
The signature of `serialize_pk_and_row` should be changed in the future. As all deletions are now bound with the original value, there's no possibility of passing a `None` row into this function.